src/Functions/DateTimeTransforms.h

```diff
@@ -624,6 +667,16 @@ struct TimeSlotImpl
         return dateIsNotSupported(name);
     }
 
+    static inline DecimalUtils::DecimalComponents<DateTime64> execute_extended_result(const DecimalUtils::DecimalComponents<DateTime64> & t, const DateLUTImpl & time_zone)
+    {
+       return {time_zone.toStartOfMinuteInterval(t.whole, 30), 0};
```

Comment:
Maybe I miss something but should l. 672 not correspond to l. 652?

Reply:
@rschu1ze This is one of the interesting points. We have:

```
struct TimeSlotImpl
{
    static inline UInt32 execute(const DecimalUtils::DecimalComponents<DateTime64> & t, const DateLUTImpl &)
    {
        return t.whole / 1800 * 1800;
    }
    
    static inline DecimalUtils::DecimalComponents<DateTime64> executeExtendedResult(const DecimalUtils::DecimalComponents<DateTime64> & t, const DateLUTImpl & time_zone)
    {
       return {time_zone.toStartOfMinuteInterval(t.whole, 30), 0};
    } 
```

So do you think that executeExtendedResult also return result without timezone? 
It should be something like this:
`return {t.whole / 1800 * 1800, 0};`

Reply:
I am working on this now. The case is interesting. For example:
`select timeSlot(toDateTime('1979-12-31 10:35:30'), 'Europe/Paris') `
returns 1979-12-31 11:30:00 (with timezone).
while:
`select timeSlot(toDateTime('1979-12-31 10:35:30', 'Europe/Paris')) `
returns without timezone: 1979-12-31 10:30:00

Probably I will implement the same for DateTime64 argument with enable_extended_results_for_datetime_functions = true.

Reply:
OK. I implement timeSlot() using more similar approach:

```
    static inline DecimalUtils::DecimalComponents<DateTime64>  executeExtendedResult(const DecimalUtils::DecimalComponents<DateTime64> & t, const DateLUTImpl &)
    {
        if (likely(t.whole >= 0))
            return {t.whole / 1800 * 1800, 0};
        return {(t.whole + 1 - 1800) / 1800 * 1800, 0};
    }
```
For negative values (dates before 1970) we still need to do things like:` t.whole + 1 - 1800) / 1800 * 1800 ` to make correct rounding. Otherwise 1969-12-31 13:35:30 will be rounded up to 1969-12-31 14:00:00 (instead of correct  1969-12-31 13:30:00 )

Reply:
@rvasin , You've checked that it works for non default scale, right?

Reply:
All this business with "+1 - 1800" seems correct. The shame is I cannot think up a helpful comment.

Reply:
Yes, I have verified: it works fine with non-default scale:
`select timeSlot(toDateTime64('1969-12-31 10:35:30.75012', 5), 'Europe/Paris')  SETTINGS enable_extended_results_for_datetime_functions = true;`
Returns: 1969-12-31 11:30:00.00000

Reply:
I am good with this implementation.